middle_end/flambda2/simplify/simplify_nullary_primitive.ml

```diff
@@ -42,8 +42,7 @@ let simplify_nullary_primitive dacc original_prim (prim : P.nullary_primitive)
     Simplify_primitive_result.create named ~try_reify:false dacc
   | Enter_inlined_apply { dbg } ->
     let dacc =
-      DA.map_denv dacc ~f:(fun denv ->
-          DE.set_inlined_debuginfo denv (DE.add_inlined_debuginfo denv dbg))
+      DA.map_denv dacc ~f:(fun denv -> DE.set_inlined_debuginfo' denv dbg)
```

Comment:
similarly to what needs to be done in `to_cmm` I think that this code should also be fixed to somehow "stack" the inlined_debuginfos, else this line will lose any previous/less deep inlined debuginfos.

Reply:
Should be fixed